### PR TITLE
Support for UUID and Unix timestamp functions in Rhai

### PR DESCRIPTION
**Pull request by lleadbet**
*Description*

When building Rhai scripts, it's often that you'll need to add headers that either uniquely identify a request, or append timestamp information for processing information later, such as crafting a trace header or otherwise. 

While the default `timestamp()` and associated functions (e.g. `apollo_start`) can be used for a somewhat similar function, it also isn't able to be translated into an epoch. As a result, it doesn't acutely address the asks we've heard from customers. 

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible
- [x] Documentation completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing
    - [x] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Notes**

* Unsure on how to measure performance impact; would be happy with information and I can add it
* Unsure how new functions in Rhai are tested, as the existing tests don't seem to have specific unit tests against it. 
* Manually tested locally using a simple Rhai script: 

```rhai
fn supergraph_service(service) {
    print(uuid());
    print(unix_now());
}
```

### CODE REVIEW

**Review by o0Ignition0o — APPROVED** [review]
LGTM, I love the documentation!

**Comment by lennyburdette** [comment]
Nice! Should the function be uuidV4 to leave room for other versions and make it obvious?

**Comment by lleadbet** [comment]
> Nice! Should the function be uuidV4 to leave room for other versions and make it obvious?

Good callout, probably should make it more obvious that it's v4 like I did with the timestamp being `unix_now` vs. just `now`. :)

**Comment by garypen** [comment]
It's always tricky to add unit tests for timing functionality. I'd probably do something like this for `unix_now()`:
```

    #[test]
    fn it_can_create_unix_now() {
        let engine = Rhai::new_rhai_engine(None);
        let st = SystemTime::now()
            .duration_since(SystemTime::UNIX_EPOCH)
            .expect("can get system time")
            .as_secs();
        let unix_now: u64 = engine
            .eval(r#"unix_now()"#)
            .expect("can get unix_now() timestamp");
        // Always difficult to do timing tests. unix_now() should execute within a second of st,
        // so...
        assert!(unix_now <= st + 1);
    }
```

For the uuid, I'd generate a uuid using my function and them make sure I could re-parse the string into a valid `uuid`. Something like:
```
const ID: Uuid = uuid!(my-generated-uuid);
```

**Comment by lleadbet** [comment]
Thanks @garypen! Added new tests for the new commands and appreciate the pointer there.

**Review by garypen — APPROVED** [review]
Thanks for the addition!

File: .changesets/feat_uuid_unix_timestamp_rhai_functions.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+### Support for UUID and Unix timestamp functions in Rhai 
+
+*Description*
+
+When building Rhai scripts, it's often that you'll need to add headers that either uniquely identify a request, or append timestamp information for processing information later, such as crafting a trace header or otherwise. 
+
+While the default `timestamp()` and similar functions (e.g. `apollo_start`) can be used for a somewhat similar function, it also isn't able to be translated into an epoch. As a result, it doesn't acutely address the asks we've heard from users.
+
+This adds a `uuid_v4()` and `unix_now()` function to obtain a UUID and Unix timestamp, respectively.
+
```
Comment (abernix) [inline .changesets/feat_uuid_unix_timestamp_rhai_functions.md:10]: Just dropping by to say that this was an **excellent** changelog entry, @lleadbet.  Thanks for that!

Comment (archanasevak) [reply .changesets/feat_uuid_unix_timestamp_rhai_functions.md:10]: hi there, I am trying to use this function under supergrpah_service but I keep getting ` service callback failed: Function not found: uuid_v4 () (line 4, position 14) in call to function supergraph_service` 
heres my script: 
```rhai
// At the supergraph_service stage, register callbacks for processing requests and
// responses.
fn supergraph_service(service) {
    let id = uuid_v4();
    print(id);
```
Is there any thing needed to import to use this function?

**Pull request merged**